### ISSUE TITLE
`\listoftodos` not working with LaTeX style files for IOS Press journal articles

### ISSUE THREAD

**Issue by daniporr** [henrikmidtiby/todonotes#73]
The `\listoftodos` command does not work with the LaTeX style files for IOS Press journal articles.
For instance, with the one you can download at https://vtex-soft.github.io/texsupport.iospress-jnr.

Minimal example:

```tex
\documentclass{iosart2x}
\usepackage{todonotes}
\begin{document}
\listoftodos
Test\todo{Testing todonotes}
\end{document}
```

It produces several errors in the ".tdo" file:

```log
./test.tdo:1: Undefined control sequence.
\@dottedtocline ...#2\relax \rightskip \@tocrmarg 
                                                  \parfillskip -\rightskip \...
l.1 ...nge}{o}}\ Testing todonotes}{1}{section*.2}
                                                  %
[...]
```

Without `\listoftodos` it works as expected.

It is worth noticing that without any `\todo` it also work as expected, and it prints the "Todo list" header (obviously with an empty list).

It is also worth noticing that none of the suggestions reported in "1.6.3 Conflicts with the amsart documentclass" (of the [Package documentation](http://mirrors.ctan.org/macros/latex/contrib/todonotes/todonotes.pdf)) work in this case.

Any suggestions on how to solve this problem?

**Comment by FrankMittelbach** [comment]
As far as I can see the class you are using is not supporting any `\tableofcontents` like commands. It therefore doesn't define a value for `\@tocrmarg` that todonotes expects to be defined (and which is defined in any standard class or in those derived from them). You could look into article.cls for how it is defined there and do that in your preamble but you might find that more is missing (I have't installed the whole class so don't know).

If it is just necessary to add such definitions then perhaps todonotes could check if there are already definitions and if not define them but perhaps the class should be altered to allow for \tableofcontents (and similar structures in the first place.

**Comment by daniporr** [comment]
Thank you for your answer.
I think that is out of my league.
I am not even sure how to define a value for `\@tocrmarg`.

However, by looking at the "iosart2x.cls" file (the one mentioned above), I noticed that it contains the following line

```tex
\setcounter{tocdepth}{3}
```

which seems weird for a class that does not support any `\tableofcontents` like commands.
I am starting to think that whoever created this class file was not really an expert of LaTeX.

**Comment by daniporr** [comment]
After playing with it, I realised that if you just add

```tex
\newcommand\@pnumwidth{1.55em}
\newcommand\@tocrmarg{2.55em}
\newcommand\@dotsep{4.5}
```

`\listoftodos` works as expected!

Therefore, there is no need of changing the class (which might be more difficult).

Can this be integrated somehow in todonotes?

**Comment by FrankMittelbach** [comment]
my point is that it would be better if this is defined in the class and probably also definitions for \tableofcontents, \listoffigures, ...
In todonotes this should be defined only if there is no definition for them, so it would need something like
```
\@ifundefined{@tocrmarg}{\def\@tocrmarg{2.55em}}
```
etc. That's something @henrikmidtiby  might be willing to do, but that's up to him.

**Comment by daniporr** [comment]
This makes sense to me. It is always useful to have "default values" for variables.

**Comment by henrikmidtiby** [comment]
@daniporr Thanks for raising the issue. I think it makes sense to address it directly in the todonotes package.

@FrankMittelbach Thanks for the suggestion of using the `@ifundefined` macro to ensure that the required values exists. I stumbled on an issue with that approach together with the `iosart2x` documentclass.

Afterwards I tried to use this construct in `todonotes.sty`, which seems to work with both the `article` and the `iosart2x` documentclasses.

```
\providecommand{\@tocrmarg}{2.55em}
\providecommand{\@dotsep}{4.5}
\providecommand{\@pnumwidth}{1.55em}
```

Are you aware of something problematic in this approach? Otherwise I will include it in the next release.
